devdocs/troubleshooting-guide.md

```diff
@@ -0,0 +1,214 @@
+
+## Troubleshooting project setup
+
+### Setup checklist
+
+Use this checklist to verify that your project setup is correct.
+
+1. **Undo any automatic setting changes done by Eclipse**
+
+    Make sure Eclipse hasn't modified any settings/files during set up. Run git status to ensure the repo is same as the one you cloned from GitHub. If there are any changes, do an git reset --hard and restart Eclipse.
+
+2. **Check the version of the Google plugin for Eclipse**
+
+    In Eclipse, go to Help → Install new software → What's already installed
+
+    Ensure that Google Plugin version is `4.3 or 4.4` and TestNG is installed.
+
+    ![setup-checklist-1.png](images/setup-checklist-1.png)
+
+3. **Check the text encoding settings**
+
+    Window → Preferences → General → Workspace
+
+    ![setup-checklist-2.png](images/setup-checklist-2.png)
+
+4. **Check the JDK compatibility**
+
+    Project → Properties → Java Compiler
+
+    ![setup-checklist-3.png](images/setup-checklist-3.png)
+
+5. **Check the SDK version**
+   ```
+    Note : We are using version 1.9.27 instead of 1.9.4
+   ```
+    Project → Properties → Google → App Engine
+
+    ![setup-checklist-4.png](images/setup-checklist-4.png)
+
+6. **Check for compilation errors**
+
+    Markers tab should have 0 errors.
+
+    ![setup-checklist-5.png](images/setup-checklist-5.png)
+
+    If it shows errors, try refreshing the project (right-click on the project in Project explorer) followed by a Project → Clean.
+
+### Common setup errors and solutions
+
+1. **ERROR**: Eclipse complains about FileWriter and ConsoleHandler e.g. java.io.FileWriter is not supported by Google App Engine's Java runtime environment.
+
+  **SOLUTION**: These are OK because the offending files belong to test driver and will not be deployed to GAE. Choose to "Quick Fix" and then choose to exclude those files from validation.
+
+2. **ERROR**: Eclipse complains "...your project must be configured to use a JDK in order to use JSP".
+
+  **SOLUTION**: This happens because Eclipse is only aware of JRE, not JDK (Compiling JSP requires the JDK). 
+  
+   Go to Window → Preferences → Java → Installed JREs. 
+   
+   You will note that a JRE path is the one selected, not a JDK path. To fix this, Click add → Standard VM, then for the JRE Path enter the path of the JRE folder inside your JDK installation folder. e.g., C:/jdk1.7/jre Now you should see all of the JARs added to the library section.
+
+3. **ERROR**: When trying to deploy, Eclipse complains "... Cannot get the System Java Compiler. Please use a JDK, not a JRE.".
+
+  **SOLUTION**: You can force Eclipse to use the JDK (instead of JRE) by modifying the eclipse.ini file. See here for more details.
```

Comment:
The reference for "here" is missing. Here it is: http://stackoverflow.com/questions/13913019/changing-jdk-in-eclipse